### ISSUE TITLE
Need to escape paths before passing them to sed.

### ISSUE THREAD

**Issue by Artem-B** [vermaden/beadm#35]
My boot filesystems are in zroot/system/root/{BE1,BE2...}
`POOL=zroot` and I `BEDS=system/root` is set in /usr/local/etc/beadm.conf

However, '/' in BEDS results in sed errors and breaks beadm functionality.
What's worse, beadm ignores the errors and proceeds running various commands with root permissions using invalid arguments. Luckily it didn't mess anything up, but it may be prudent to add "set -e" at the beginning of the script to catch cases when things go wrong.

I've worked the issue around by explicitly escaping '/' in  beadm.conf:
`BEDS='system\/root'`

**Comment by vermaden** [comment]
Hi,

thanks for finding that ... but I think I would modify beadm(8) in other way.

The only valid chars for BE name are: **a-z A-Z 0-9 and - _ . ,**

Thus I would use '+' as sed(1) separator instead of currently used '/' character:

So this:

```
sed -E s/"^${POOL}\/${BEDS}\/${SOURCE##*/}"/"${POOL}\/${BEDS}\/${2##*/}"/g )
```

Will become this:

```
sed -E s+"^${POOL}\/${BEDS}\/${SOURCE##*/}"+"${POOL}\/${BEDS}\/${2##*/}"+g )
```

Can you test that?

Thanks.

**Comment by Artem-B** [comment]
That would work, too. I'm fine with whatever does the job. :-)
`@` might be a bit better choice for the separator, as `+` would be somewhat confusing considering that we're dealing with regex arguments here.

**Comment by vermaden** [comment]
Please test this version then:
https://pastebin.com/vEnX9X0v

**Comment by Artem-B** [comment]
Tested your version on FreeBSD-12.2. Works fine.

**Comment by vermaden** [comment]
Ok, I will commit that change.

Thanks for help and testing.

Regards.

**Comment by Artem-B** [comment]
Fixed by 14f998a9e132a2fdc30e62a60d840ae836d76217
